### PR TITLE
fix!: remove padding from decryptedBytes if left in by the cipher

### PR DESCRIPTION
**Pull request by andrejborstnik**
On (some?) Pixel devices (tested Pixel 6 and Pixel 7 on Android 14, `AP2A.240705.004` build) for certain string lengths (e.g. `256`) the encryption/decryption using `CipherStorageKeystoreAesCbc` doesn't erase the `PKCS7` padding from the returned string.

To work around this issue I added a method that will detect & delete the padding if left in.

This technically a breaking change for any stored strings that end with a valid `PKCS7` padding of characters up to with an ascii int value of `IV.IV_LENGTH`. Not sure how likely this use-case would be.

Opened this workaround as a discussion point, happy to discuss alternatives.

Also reported to Android https://issuetracker.google.com/issues/357896946, so they can look at the issue on their side

### CODE REVIEW

File: android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreAesCbc.java [hunk 1]
```diff
@@ -228,23 +228,49 @@ protected Key generateKey(@NonNull final KeyGenParameterSpec spec) throws Genera
...
-      // decrypt the bytes using cipher.doFinal(). Using a CipherInputStream for decryption has historically led to issues
+      // decrypt the bytes using cipher.doFinal(). Using a CipherInputStream for
+      // decryption has historically led to issues
       // on the Pixel family of devices.
       // see https://github.com/oblador/react-native-keychain/issues/383
-      byte[] decryptedBytes = cipher.doFinal(bytes, IV.IV_LENGTH, bytes.length - IV.IV_LENGTH);
+      byte[] _decryptedBytes = cipher.doFinal(bytes, IV.IV_LENGTH, bytes.length - IV.IV_LENGTH);
+
+      // removing padding is required to work around a decryption padding issue with
+      // some Pixel devices e.g. for strings of length 256
```
Comment (andrejborstnik) [inline android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreAesCbc.java:271]: Based on my testing up to 2000 bytes the problematic strings lengths are
```
528 -> 543
800 -> 815
1280 -> 1295
1552 -> 1567
1824 -> 1839

```

**Comment by jahead** [comment]
would this be the reason we are getting 
* https://github.com/oblador/react-native-keychain/issues/372
* https://github.com/oblador/react-native-keychain/issues/567
* and a number of others?

**Comment by andrejborstnik** [comment]
> would this be the reason we are getting
> 
> * [Retrieving password on android fails with Caused by: android.security.KeyStoreException: Unsupported key size #372](https://github.com/oblador/react-native-keychain/issues/372)
> * [`E_CRYPTO_FAILED` on Android 13 and Pixel phones. AES crashing out while decrypting. #567](https://github.com/oblador/react-native-keychain/issues/567)
> * and a number of others?

I have only seen the issue in the OP on Pixel devices and `E_CRYPTO_FAILED` happens on other devices as well. So I would be sceptical this is the only reason. But I would not exclude this being the reason for at least some of the decryption issues, e.g. [this padding one](https://github.com/oblador/react-native-keychain/issues/567#issuecomment-1276762307) seems related on the surface

**Comment by DorianMazur** [comment]
@andrejborstnik @jahead Consider using AES GCM #678

**Pull request closed without merging**